src/components/SettingsButton/SettingsButton.js

```diff
@@ -0,0 +1,94 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { Motion, spring } from 'react-motion';
+import styled from 'styled-components';
+import IconBase from 'react-icons-kit';
+import { settings } from 'react-icons-kit/feather/settings';
+import { COLORS } from '../../constants';
+
+import * as actions from '../../actions';
+
+type Props = {
+  size: number,
+  color: ?string,
+  hoverColor: ?string,
+  settingsFor: 'project' | 'app',
+};
+
+type State = {
+  rotations: number,
+  scale: number,
+  color: number,
+  hovered: boolean,
+};
+
+class SettingsButton extends Component<Props, State> {
+  static defaultProps = {
+    size: 30,
+    color: COLORS.gray[600],
+    hoverColor: COLORS.purple[700], // purple or violet 500/700
+    settingsFor: 'project',
+  };
+
+  state = {
+    hovered: false,
+  };
+
+  handleMouseEnter = () => {
+    this.setState(state => ({
+      hovered: true,
+    }));
+  };
+
+  handleMouseLeave = () => {
+    // I wonder if we should "unwind" it on mouseout?
+    // I'm not sure... maybe try it with/without this?
+    this.setState({ hovered: false });
+  };
+
+  handleShowModal = () => {
+    const { settingsFor, showModal } = this.props;
+    showModal(settingsFor);
+  };
+
+  render() {
+    const { hovered } = this.state;
+    return (
+      // Omitting most of the structure that isn't relevant
```

Comment:
What does this comment mean, what structure is omitted?

Reply:
mmhhh, not sure. I think that was a comment that was in your proposal. 😄 
Maybe this is outdated. I'll remove it.